components/hab/mac/README.md

```diff
@@ -1,26 +1,74 @@
-# Building a hab Mac Binary
+# Building a `hab` MacOS Binary
 
-As Habitat currently does not have first class support for the Mac platform, a pragmatic approach has been taken to build a `hab` binary for Mac OS X. A wrapper script called `mac-build.sh` attempts to install any missing pre-requisites before invoking the `hab-plan-build.sh` program with a custom `PATH`. Currently, the following are required on the Mac system performing the build:
+As Habitat currently does not have first class support for the Mac platform, a pragmatic approach has been taken to build a `hab` binary for MacOS. This details the steps to build a release on MacOS. It is also currently codified in [.buildkite/scripts/build_mac_release.sh](habitat-sh/habitat/.buildkite/scripts/build_mac_release.sh)
 
-* Xcode CLI tools
-* Homebrew
-* `coreutils`, `gnu-tar`, and `wget` Homebrew packages
-* Rust
-* Cargo nightly
+## Prerequisites
 
-A final prerequisite of a `hab` binary itself is required by `hab-plan-build.sh` to sign the resulting Habitat artifact. The `mac-build.sh` will install the latest release from Bintray if `/bin/hab` cannot be found locally, otherwise the `hab` on your `PATH` will be used. As we are using the build program, a secret origin key is also required in the Mac's key cache under `/hab/cache/keys`.
-
-## Usage
+### Install XCode Command Line Developer Tools
+This installs basic developer tooling needed to compile and build software on a Mac.
 
 ```sh
-cd components/hab/mac
-sudo ./mac-build.sh
+xcode-select --install
 ```
 
-Assuming success, this will produce a local `./results` directory with the artifact.
+### Install Omnibus Bootstrap Toolchain package
+
+Since there is not yet a complete Habitat build toolchain available for MacOS, we provide the minimal set of binaries and static libraries needed to compile a `hab` binary using Chef's Omnibus tooling platform. This effectively takes the place of the packages we would add to a `pkg_build_deps` entry in a Habitat plan file.
 
-Alternatively, as `mac-build.sh` is a wrapper around the build program, it can be just as easily invoked from the root of the source tree with:
+TODO: Where can this package be retrieved from?
 
 ```sh
-sudo ./components/hab/mac/mac-build.sh components/hab/mac
+sudo installer \
+     -pkg <PATH_TO_BOOTSTRAP_TOOLCHAIN_PACKAGE> \
+     -target /
+```
+
+TODO: Can we rename `hab-bundle`?
+
+This will install the toolchain in `/opt/hab-bundle`; this is where the build program is expecting to find binaries and libraries it needs.
+
+### Install Homebrew
+Follow the instructions at https://brew.sh.
+
+### Install Rust toolchain
+Follow the instructions at https://rustup.rs/
+
+### Install Homebrew prerequisites
```

Comment:
I think that in light of these changes we may want to consider updating our Mac/Development instructions to use similar deps when making development `cargo build`s. That is, use the same Omnibus/Homebrew combinations in development so that we more closely match a production build later on.

Reply:
This is going to be a work in progress as we tweak the process and automate more bits - but yes you are totally correct.